src/routes/competition/FilterBar.svelte

```diff
@@ -0,0 +1,64 @@
+<script lang="ts">
+	import { Input } from 'flowbite-svelte';
+	import { debounce } from '../../utils/debounce';
+	import type { League } from '../../types/League';
+	import type { Nation } from '../../types/Nation';
+	import type { Season } from '../../types/Season';
+	import SwitchButton from '../../compnonents/shared/SwitchButton.svelte';
+	import SelectFilter from '../../compnonents/competitions/SelectFilter.svelte';
+
+	export let viewCalendar: boolean;
+	export let name: string;
+	export let year: number;
+	export let league: string;
+	export let nation: string;
+	export let seasons: Season[];
+	export let nations: Nation[];
+	export let leagues: League[];
+
+	const handleKeyup = async (target: EventTarget | null) => {
+		if (!(target instanceof HTMLInputElement)) return;
+		await debounce();
```

Comment:
```suggestion
		if(!await debounce()) return;
```
This will have to be changed to accommodate for the changes in #28

Reply:
Yes, we can merge the debounce PR and then pull the changes in here